config/yaml.go

```diff
@@ -76,40 +113,29 @@ func NewYAMLProviderFromFiles(mustExist bool, resolver FileResolver, files ...st
 	return newYAMLProviderCore(readers...)
 }
 
-// NewYamlProviderFromReader creates a configuration provider from an
-// io.ReadCloser
-func NewYamlProviderFromReader(reader io.ReadCloser) Provider {
-	return newYAMLProviderCore(reader)
+// NewYamlProviderFromReader creates a configuration provider from a list of `io.ReadClosers`.
+// As above, all the objects are going to be merged and arrays/values overridden in the order of the files.
+func NewYamlProviderFromReader(readers ...io.ReadCloser) Provider {
+	return newYAMLProviderCore(readers...)
 }
 
-// NewYAMLProviderFromBytes creates a config provider from a byte-backed YAML
-// blob.
-func NewYAMLProviderFromBytes(yaml []byte) Provider {
-	reader := bytes.NewReader(yaml)
-	node, err := newyamlNode(ioutil.NopCloser(reader))
-	if err != nil {
-		panic(err)
+// NewYAMLProviderFromBytes creates a config provider from a byte-backed YAML blobs.
+// As above, all the objects are going to be merged and arrays/values overridden in the order of the yamls.
+func NewYAMLProviderFromBytes(yamls ...[]byte) Provider {
+	closers := make([]io.ReadCloser, len(yamls))
+	for i := range yamls {
```

Comment:
Hmm, I thought, it is going to copy the temp variable array and use later, but it is not. Thanks!